src/__tests__/resources/timeSeriesObject.unit.spec.ts

```diff
@@ -0,0 +1,130 @@
+import { TimeSeries } from '../../resources/classes/timeSeries';
+import { TimeSeriesList } from '../../resources/classes/timeSeriesList';
+import { randomInt, setupLoggedInClient } from '../testUtils';
+
+describe('TimeSeries class unit test', () => {
+  let axiosMock: MockAdapter;
+  let client: CogniteClient;
+  let newTimeSeries: any;
+  let newAsset: any;
+  let createdTimeSeries: TimeSeriesList;
+  let timeSeriesWithAssetId: TimeSeries;
+  let createdAssets: AssetList;
+  let datapointArray: any[] = [];
+  beforeAll(() => {
+    client = setupLoggedInClient();
+    axiosMock = new MockAdapter(client.instance);
+  });
+  beforeEach(async () => {
+    axiosMock.reset();
+
+    // Create assets
+    newAsset = {
+      name: 'test-asset' + randomInt(),
+      externalId: 'asset' + randomInt(),
+      id: randomInt(),
+    };
+    axiosMock
+      .onPost(new RegExp('/assets$'), { items: [newAsset] })
+      .replyOnce(200, { items: [newAsset] });
+    createdAssets = await client.assets.create([newAsset]);
+
+    // Create timeseries
+    newTimeSeries = {
+      name: 'test-timeseries',
+      externalId: 'timeseries' + randomInt(),
+      id: randomInt(),
+    };
+    timeSeriesWithAssetId = {
+      ...newTimeSeries,
+      assetId: createdAssets[0].id,
+      externalId: 'timeseriesWithAssetId' + randomInt(),
+      id: randomInt(),
+    };
+    axiosMock
+      .onPost(new RegExp('/timeseries$'), {
+        items: [newTimeSeries, timeSeriesWithAssetId],
+      })
+      .replyOnce(200, { items: [newTimeSeries, timeSeriesWithAssetId] });
+    createdTimeSeries = await client.timeseries.create([
+      newTimeSeries,
+      timeSeriesWithAssetId,
+    ]);
+
+    // Create datapoints
+    datapointArray = [];
+    const time = randomInt();
+    for (let index = 0; index < 3; index++) {
+      const datapoint = {
+        id: createdTimeSeries[1].id,
+        datapoints: [{ timestamp: time + index, value: 10 + index }],
+      };
+      datapointArray.push(datapoint);
+    }
+    axiosMock
+      .onPost(new RegExp('/timeseries/data$'), {
+        items: datapointArray,
+      })
+      .replyOnce(200);
+    await client.datapoints.insert(datapointArray);
+  });
+
+  test('create', async () => {
+    expect(createdTimeSeries[0]).toBeInstanceOf(TimeSeries);
+    expect(createdTimeSeries[0].externalId).toEqual(newTimeSeries.externalId);
+    expect(createdTimeSeries[1]).toBeInstanceOf(TimeSeries);
+    expect(createdTimeSeries[1].externalId).toEqual(
+      timeSeriesWithAssetId.externalId
+    );
+    expect(createdTimeSeries[1].assetId).toBeDefined();
+  });
+
+  test('get asset', async () => {
+    axiosMock
+      .onPost(new RegExp('/assets/byids$'), {
+        items: [{ id: createdTimeSeries[1].assetId }],
+      })
+      .replyOnce(200, { items: [newAsset] });
+    const assetsFromTimeseries = await createdTimeSeries[1].getAsset();
+    if (assetsFromTimeseries) {
+      expect(assetsFromTimeseries[0]).toBeInstanceOf(Asset);
```

Comment:
I think it is strange we return an array here. Since if you have a single timeseries then it can maximum have 1 asset.

Reply:
Good point, it is fixed now